pkg/chunk/cache/memcached_client.go

```diff
@@ -114,17 +140,25 @@ func (c *memcachedClient) updateLoop(updateInterval time.Duration) {
 // updateMemcacheServers sets a memcache server list from SRV records. SRV
 // priority & weight are ignored.
 func (c *memcachedClient) updateMemcacheServers() error {
-	_, addrs, err := net.LookupSRV(c.service, "tcp", c.hostname)
-	if err != nil {
-		return err
-	}
 	var servers []string
-	for _, srv := range addrs {
-		servers = append(servers, fmt.Sprintf("%s:%d", srv.Target, srv.Port))
+
+	if len(c.addresses) > 0 {
+		c.provider.Resolve(context.Background(), c.addresses)
```

Comment:
Maybe worth to add a sane hardcoded timeout to the context, to avoid the  request stuck for any reason. Even an high timeout (10s) will do the job.